### DISCUSSION TITLE
did not find expected authorization request details in session

### DISCUSSION THREAD

**Discussion by awesomelike** [panva/openid-client#435] [Q&A] [answered]
**Describe the bug**

I am using Passport strategy of this library, with express-session Redis store.
`did not find expected authorization request details in session ` when the app is deployed. I have set the correct domain for cookie settings.
What are the possible causes of this problem?

**Comment by awesomelike** [comment]
Everything works fine locally, but when deployed, this error shows up.
I am not sure what is broken

**Reply by panva** [reply]
- Maybe the way your session cookie attributes are set up?
- Maybe you're using response mode form_post and your cookies are not sameSite=none? E.g. they're not going to be attached to the callback POST from the issuer?
- Maybe you're using some redis replication and are reading from a secondary that is yet to be replicated to?
- ...

I'd start by checking the session is actually present before calling passport and inspecting the network tab in your browser's developer tools looking at the set-cookie and cookie headers in requests/responses.

**Reply by awesomelike** [reply]
Thank you for the help

**Comment by awesomelike** [comment, marked as answer]
My application server was deployed behind AWS ALB.
I set `app.set('trust proxy', 1);` and the error has disappeared.

**Reply by yiqun8313** [reply]
it doesn't works
